src/lighthouse.rs

```diff
@@ -89,56 +112,94 @@ fn quorum_changed(a: &Vec<QuorumMember>, b: &Vec<QuorumMember>) -> bool {
     return a_ids != b_ids;
 }
 
-// Checks whether the quorum is valid and an explanation for the state.
-fn quorum_valid(state: &RoomState, opt: &LighthouseOpt) -> (bool, String) {
-    let mut first_joined = Instant::now();
+// Checks whether the quorum is valid, the new quorum and an explanation for the state.
+fn quorum_compute(
+    now: Instant,
+    heartbeats: &HashMap<String, Instant>,
+    state: &RoomState,
+    opt: &LighthouseOpt,
+) -> (Option<Vec<QuorumMember>>, String) {
+    let healthy_participants: HashMap<String, QuorumMemberDetails> = state
```

Comment:
perhaps it might be worth refactoring calculating healthy participants to a separate helper? Then the logic for quorum could be a bit simpler:

health checker component -- maintains the state of all healthy participants

quorum:
- uses the health checker component to see which participants are alive and their states

Reply:
That's a fair point, we could pull it out but we still need some of this info in the compute_quorum itself. In a follow up PR we added a "split brain" prevention so we won't create a quorum with less than half of the known healthy workers

We also use it for logging purposes